charmbits/httpservice/service.go

```diff
@@ -210,68 +209,102 @@ func (svc *Service) Restart() error {
 }
 
 type server struct {
+	tomb    tomb.Tomb
 	handler *handler
+
+	httpListener  net.Listener
+	httpsListener net.Listener
 }
 
-func (svc *Service) startServer(ctxt *service.Context, args []string) {
+func (svc *Service) startServer(ctxt *service.Context, args []string) (hook.Command, error) {
 	srv := server{
 		handler: svc.handler,
 	}
-	if err := srv.start(ctxt, args); err != nil {
-		fmt.Fprintf(os.Stderr, "%s\n", err)
-		os.Exit(1)
-	}
+	return srv.start(ctxt, args)
 }
 
-func (srv *server) start(ctxt *service.Context, args []string) error {
+func (srv *server) start(ctxt *service.Context, args []string) (hook.Command, error) {
 	if len(args) != 4 {
-		return errgo.Newf("got %d arguments, expected 2", len(args))
+		return nil, errgo.Newf("got %d arguments, expected 2", len(args))
```

Comment:
this somehow seems confusing.. above there's if len(args) != 4, but here it says we expect 2 arguments..